tests/generation/SettingsClassTest.py

```diff
@@ -0,0 +1,90 @@
+import unittest
+
+from pyprint.ConsolePrinter import ConsolePrinter
+from coala_quickstart.generation.SettingsClass import (
+    collect_bear_settings, BearSettings)
+from tests.test_bears.AllKindsOfSettingsDependentBear import (
+    AllKindsOfSettingsDependentBear)
+from tests.test_bears.AllKindsOfSettingsDependentDecoratedBear import (
+    AllKindsOfSettingsDependentDecoratedBear)
+from tests.test_bears.AllKindsOfSettingsBaseBear import (
+    AllKindsOfSettingsBaseBear)
+from tests.test_bears.BearA import BearA
+
+
+class TestSettingsClass(unittest.TestCase):
+
+    def setUp(self):
+        self.printer = ConsolePrinter()
+        self.log_printer = None
+
+    def test_collect_bear_settings(self):
+        relevant_bears = {'test':
+                          {AllKindsOfSettingsDependentBear,
+                           AllKindsOfSettingsDependentDecoratedBear,
+                           AllKindsOfSettingsBaseBear,
+                           BearA}}
+
+        bear_settings_obj = collect_bear_settings(relevant_bears)
+
+        k = 0
+        for index, i in enumerate(bear_settings_obj):
+            if i.bear == AllKindsOfSettingsDependentBear:
+                k = index
+                break
+        obj = bear_settings_obj[k].non_optional_settings
+        self.assertCountEqual(obj.settings_bool, ['use_bear', 'use_bears'])
```

Comment:
`assertCountEqual` can be misleading.  Both the lists need to match exactly. Please use `assertEqual`.

Reply:
Same for other occurences of `assertCountEqual` in this file.

Reply:
well, the list order is not fixed for the settings, I can sort both the lists and then compare but `assertCountEqual` achives the same thing. Do you want the order of the list to be fixed? I can sort the list after populating the class is complete although I don't see whether it would be of any advantage, as we will be just launching these bears all in parallel and it would be futile to use the sorted list fact and try launching the bears with different settings a priori to the others

Reply:
> Do you want the order of the list to be fixed? 

Ah no, I thought `assertCountEqual` just compares the length, didn't knew it matches the elements as well.